qa: fix test_mirroring_init_failure_with_recovery failed to get state: "failed"

joscollin opened the pull request:
Fixes: https://tracker.ceph.com/issues/68187






<!--
  - Please give your pull request a title like

      [component]: [short description]

  - Please use this format for each git commit message:

      [component]: [short description]

      [A longer multiline description]

      Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
      Signed-off-by: [Your Name] <[your email]>

    For examples, use "git log".
-->

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

joscollin commented:
> @joscollin status of this?

@vshankar Fix under testing. Will update soon.

vshankar commented:
@joscollin status of this? Seeing again in https://pulpito.ceph.com/vshankar-2025-02-27_13:15:04-fs-wip-vshankar-testing-20250227.075314-debug-testing-default-smithi/8158468/

joscollin commented:
> @joscollin status of this? Seeing again in https://pulpito.ceph.com/vshankar-2025-02-27_13:15:04-fs-wip-vshankar-testing-20250227.075314-debug-testing-default-smithi/8158468/

@vshankar 
I don't see this failure coming in my latest tests, especially with the fix https://github.com/ceph/ceph/pull/61539.

Checking the logs, I see:
```
2025-02-27T16:07:49.928 WARNING:tasks.cephfs.test_mirroring:mirror daemon command with label "mirror status for fs: cephfs" failed: Command failed (mirror status for fs: cephfs) on smithi097 with status 22: 'ceph --admin-daemon /var/run/ceph/cephfs-mirror.asok fs mirror status cephfs@62'
2025-02-27T16:07:54.929 INFO:teuthology.orchestra.run:Running command with timeout 30
2025-02-27T16:07:54.930 DEBUG:teuthology.orchestra.run.smithi097:mirror status for fs: cephfs> ceph --admin-daemon /var/run/ceph/cephfs-mirror.asok fs mirror status cephfs@62
2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:no valid command found; 10 closest matches:
2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:0
2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:1
2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:2
2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:abort
2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:assert
2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config diff
2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config diff get <var>
2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config get <var>
2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config help [<var>]
2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config set <var> <val>...
2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:admin_socket: invalid command
2025-02-27T16:07:55.092 DEBUG:teuthology.orchestra.run:got remote process result: 22
```
This^ should not happen.
https://github.com/ceph/ceph/blob/main/qa/tasks/cephfs/test_mirroring.py#L898-L901 - This doesn't make the mirroring restarted and the command `fs mirror status` fails.

vshankar commented:
> > @joscollin status of this? Seeing again in https://pulpito.ceph.com/vshankar-2025-02-27_13:15:04-fs-wip-vshankar-testing-20250227.075314-debug-testing-default-smithi/8158468/
> 
> @vshankar I don't see this failure coming in my latest tests, especially with the fix #61539.

Could you explain how #61539 fixes this? I don't see an obvious relation.

> 
> Checking the logs, I see:
> 
> ```
> 2025-02-27T16:07:49.928 WARNING:tasks.cephfs.test_mirroring:mirror daemon command with label "mirror status for fs: cephfs" failed: Command failed (mirror status for fs: cephfs) on smithi097 with status 22: 'ceph --admin-daemon /var/run/ceph/cephfs-mirror.asok fs mirror status cephfs@62'
> 2025-02-27T16:07:54.929 INFO:teuthology.orchestra.run:Running command with timeout 30
> 2025-02-27T16:07:54.930 DEBUG:teuthology.orchestra.run.smithi097:mirror status for fs: cephfs> ceph --admin-daemon /var/run/ceph/cephfs-mirror.asok fs mirror status cephfs@62
> 2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:no valid command found; 10 closest matches:
> 2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:0
> 2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:1
> 2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:2
> 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:abort
> 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:assert
> 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config diff
> 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config diff get <var>
> 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config get <var>
> 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config help [<var>]
> 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config set <var> <val>...
> 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:admin_socket: invalid command
> 2025-02-27T16:07:55.092 DEBUG:teuthology.orchestra.run:got remote process result: 22
> ```
> 
> This^ should not happen. https://github.com/ceph/ceph/blob/main/qa/tasks/cephfs/test_mirroring.py#L898-L901 - This doesn't make the mirroring restarted and the command `fs mirror status` fails.

This means the mirroring commands are not registered. Is the mgr undergoing restarts?

joscollin commented:
> > > @joscollin status of this? Seeing again in https://pulpito.ceph.com/vshankar-2025-02-27_13:15:04-fs-wip-vshankar-testing-20250227.075314-debug-testing-default-smithi/8158468/
> > 
> > 
> > @vshankar I don't see this failure coming in my latest tests, especially with the fix #61539.
> 
> Could you explain how #61539 fixes this? I don't see an obvious relation.

I didn't say that #61539 fixes this issue.
I meant, I don't see this failure coming in my latest tests.

> 
> > Checking the logs, I see:
> > ```
> > 2025-02-27T16:07:49.928 WARNING:tasks.cephfs.test_mirroring:mirror daemon command with label "mirror status for fs: cephfs" failed: Command failed (mirror status for fs: cephfs) on smithi097 with status 22: 'ceph --admin-daemon /var/run/ceph/cephfs-mirror.asok fs mirror status cephfs@62'
> > 2025-02-27T16:07:54.929 INFO:teuthology.orchestra.run:Running command with timeout 30
> > 2025-02-27T16:07:54.930 DEBUG:teuthology.orchestra.run.smithi097:mirror status for fs: cephfs> ceph --admin-daemon /var/run/ceph/cephfs-mirror.asok fs mirror status cephfs@62
> > 2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:no valid command found; 10 closest matches:
> > 2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:0
> > 2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:1
> > 2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:2
> > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:abort
> > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:assert
> > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config diff
> > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config diff get <var>
> > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config get <var>
> > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config help [<var>]
> > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config set <var> <val>...
> > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:admin_socket: invalid command
> > 2025-02-27T16:07:55.092 DEBUG:teuthology.orchestra.run:got remote process result: 22
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > This^ should not happen. https://github.com/ceph/ceph/blob/main/qa/tasks/cephfs/test_mirroring.py#L898-L901 - This doesn't make the mirroring restarted and the command `fs mirror status` fails.
> 
> This means the mirroring commands are not registered. Is the mgr undergoing restarts?

Yes, the mgr loads modules during that time. It doesn't seems responding to mirror commands during that time.

vshankar commented:
> > > > @joscollin status of this? Seeing again in https://pulpito.ceph.com/vshankar-2025-02-27_13:15:04-fs-wip-vshankar-testing-20250227.075314-debug-testing-default-smithi/8158468/
> > > 
> > > 
> > > @vshankar I don't see this failure coming in my latest tests, especially with the fix #61539.
> > 
> > 
> > Could you explain how #61539 fixes this? I don't see an obvious relation.
> 
> I didn't say that #61539 fixes this issue. I meant, I don't see this failure coming in my latest tests.

The "especially with fix ..." sounded like that PR fixes something under-the-hood.

> 
> > > Checking the logs, I see:
> > > ```
> > > 2025-02-27T16:07:49.928 WARNING:tasks.cephfs.test_mirroring:mirror daemon command with label "mirror status for fs: cephfs" failed: Command failed (mirror status for fs: cephfs) on smithi097 with status 22: 'ceph --admin-daemon /var/run/ceph/cephfs-mirror.asok fs mirror status cephfs@62'
> > > 2025-02-27T16:07:54.929 INFO:teuthology.orchestra.run:Running command with timeout 30
> > > 2025-02-27T16:07:54.930 DEBUG:teuthology.orchestra.run.smithi097:mirror status for fs: cephfs> ceph --admin-daemon /var/run/ceph/cephfs-mirror.asok fs mirror status cephfs@62
> > > 2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:no valid command found; 10 closest matches:
> > > 2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:0
> > > 2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:1
> > > 2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:2
> > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:abort
> > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:assert
> > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config diff
> > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config diff get <var>
> > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config get <var>
> > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config help [<var>]
> > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config set <var> <val>...
> > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:admin_socket: invalid command
> > > 2025-02-27T16:07:55.092 DEBUG:teuthology.orchestra.run:got remote process result: 22
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > This^ should not happen. https://github.com/ceph/ceph/blob/main/qa/tasks/cephfs/test_mirroring.py#L898-L901 - This doesn't make the mirroring restarted and the command `fs mirror status` fails.
> > 
> > 
> > This means the mirroring commands are not registered. Is the mgr undergoing restarts?
> 
> Yes, the mgr loads modules during that time. It doesn't seems responding to mirror commands during that time.

But `fs mirror ...` commands are run inside safe_while(), so it would retry till it times out. Did the test hit the max retry failure?

joscollin commented:
> > > > > @joscollin status of this? Seeing again in https://pulpito.ceph.com/vshankar-2025-02-27_13:15:04-fs-wip-vshankar-testing-20250227.075314-debug-testing-default-smithi/8158468/
> > > > 
> > > > 
> > > > @vshankar I don't see this failure coming in my latest tests, especially with the fix #61539.
> > > 
> > > 
> > > Could you explain how #61539 fixes this? I don't see an obvious relation.
> > 
> > 
> > I didn't say that #61539 fixes this issue. I meant, I don't see this failure coming in my latest tests.
> 
> The "especially with fix ..." sounded like that PR fixes something under-the-hood.
> 
> > > > Checking the logs, I see:
> > > > ```
> > > > 2025-02-27T16:07:49.928 WARNING:tasks.cephfs.test_mirroring:mirror daemon command with label "mirror status for fs: cephfs" failed: Command failed (mirror status for fs: cephfs) on smithi097 with status 22: 'ceph --admin-daemon /var/run/ceph/cephfs-mirror.asok fs mirror status cephfs@62'
> > > > 2025-02-27T16:07:54.929 INFO:teuthology.orchestra.run:Running command with timeout 30
> > > > 2025-02-27T16:07:54.930 DEBUG:teuthology.orchestra.run.smithi097:mirror status for fs: cephfs> ceph --admin-daemon /var/run/ceph/cephfs-mirror.asok fs mirror status cephfs@62
> > > > 2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:no valid command found; 10 closest matches:
> > > > 2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:0
> > > > 2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:1
> > > > 2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:2
> > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:abort
> > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:assert
> > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config diff
> > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config diff get <var>
> > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config get <var>
> > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config help [<var>]
> > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config set <var> <val>...
> > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:admin_socket: invalid command
> > > > 2025-02-27T16:07:55.092 DEBUG:teuthology.orchestra.run:got remote process result: 22
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > This^ should not happen. https://github.com/ceph/ceph/blob/main/qa/tasks/cephfs/test_mirroring.py#L898-L901 - This doesn't make the mirroring restarted and the command `fs mirror status` fails.
> > > 
> > > 
> > > This means the mirroring commands are not registered. Is the mgr undergoing restarts?
> > 
> > 
> > Yes, the mgr loads modules during that time. It doesn't seems responding to mirror commands during that time.
> 
> But `fs mirror ...` commands are run inside safe_while(), so it would retry till it times out. Did the test hit the max retry failure?

Yes, it hit the max retry failure.

joscollin commented:
> > > > > > > @joscollin status of this? Seeing again in https://pulpito.ceph.com/vshankar-2025-02-27_13:15:04-fs-wip-vshankar-testing-20250227.075314-debug-testing-default-smithi/8158468/
> > > > > > 
> > > > > > 
> > > > > > @vshankar I don't see this failure coming in my latest tests, especially with the fix #61539.
> > > > > 
> > > > > 
> > > > > Could you explain how #61539 fixes this? I don't see an obvious relation.
> > > > 
> > > > 
> > > > I didn't say that #61539 fixes this issue. I meant, I don't see this failure coming in my latest tests.
> > > 
> > > 
> > > The "especially with fix ..." sounded like that PR fixes something under-the-hood.
> > > > > > Checking the logs, I see:
> > > > > > ```
> > > > > > 2025-02-27T16:07:49.928 WARNING:tasks.cephfs.test_mirroring:mirror daemon command with label "mirror status for fs: cephfs" failed: Command failed (mirror status for fs: cephfs) on smithi097 with status 22: 'ceph --admin-daemon /var/run/ceph/cephfs-mirror.asok fs mirror status cephfs@62'
> > > > > > 2025-02-27T16:07:54.929 INFO:teuthology.orchestra.run:Running command with timeout 30
> > > > > > 2025-02-27T16:07:54.930 DEBUG:teuthology.orchestra.run.smithi097:mirror status for fs: cephfs> ceph --admin-daemon /var/run/ceph/cephfs-mirror.asok fs mirror status cephfs@62
> > > > > > 2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:no valid command found; 10 closest matches:
> > > > > > 2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:0
> > > > > > 2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:1
> > > > > > 2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:2
> > > > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:abort
> > > > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:assert
> > > > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config diff
> > > > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config diff get <var>
> > > > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config get <var>
> > > > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config help [<var>]
> > > > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config set <var> <val>...
> > > > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:admin_socket: invalid command
> > > > > > 2025-02-27T16:07:55.092 DEBUG:teuthology.orchestra.run:got remote process result: 22
> > > > > > ```
> > > > > > 
> > > > > > 
> > > > > >     
> > > > > >       
> > > > > >     
> > > > > > 
> > > > > >       
> > > > > >     
> > > > > > 
> > > > > >     
> > > > > >   
> > > > > > This^ should not happen. https://github.com/ceph/ceph/blob/main/qa/tasks/cephfs/test_mirroring.py#L898-L901 - This doesn't make the mirroring restarted and the command `fs mirror status` fails.
> > > > > 
> > > > > 
> > > > > This means the mirroring commands are not registered. Is the mgr undergoing restarts?
> > > > 
> > > > 
> > > > Yes, the mgr loads modules during that time. It doesn't seems responding to mirror commands during that time.
> > > 
> > > 
> > > But `fs mirror ...` commands are run inside safe_while(), so it would retry till it times out. Did the test hit the max retry failure?
> > 
> > 
> > Yes, it hit the max retry failure.
> 
> OK. In that case, if the timeout was due to ceph-mgr not responding to commands, then increasing the timeout could help in the interim. But, ceph-mgr not responding for such long is a cause of concern.

I can increase the timeout, but that's not the root cause. The root cause seems unrelated to mirroring.

vshankar commented:
> > > > > > > > @joscollin status of this? Seeing again in https://pulpito.ceph.com/vshankar-2025-02-27_13:15:04-fs-wip-vshankar-testing-20250227.075314-debug-testing-default-smithi/8158468/
> > > > > > > 
> > > > > > > 
> > > > > > > @vshankar I don't see this failure coming in my latest tests, especially with the fix #61539.
> > > > > > 
> > > > > > 
> > > > > > Could you explain how #61539 fixes this? I don't see an obvious relation.
> > > > > 
> > > > > 
> > > > > I didn't say that #61539 fixes this issue. I meant, I don't see this failure coming in my latest tests.
> > > > 
> > > > 
> > > > The "especially with fix ..." sounded like that PR fixes something under-the-hood.
> > > > > > > Checking the logs, I see:
> > > > > > > ```
> > > > > > > 2025-02-27T16:07:49.928 WARNING:tasks.cephfs.test_mirroring:mirror daemon command with label "mirror status for fs: cephfs" failed: Command failed (mirror status for fs: cephfs) on smithi097 with status 22: 'ceph --admin-daemon /var/run/ceph/cephfs-mirror.asok fs mirror status cephfs@62'
> > > > > > > 2025-02-27T16:07:54.929 INFO:teuthology.orchestra.run:Running command with timeout 30
> > > > > > > 2025-02-27T16:07:54.930 DEBUG:teuthology.orchestra.run.smithi097:mirror status for fs: cephfs> ceph --admin-daemon /var/run/ceph/cephfs-mirror.asok fs mirror status cephfs@62
> > > > > > > 2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:no valid command found; 10 closest matches:
> > > > > > > 2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:0
> > > > > > > 2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:1
> > > > > > > 2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:2
> > > > > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:abort
> > > > > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:assert
> > > > > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config diff
> > > > > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config diff get <var>
> > > > > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config get <var>
> > > > > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config help [<var>]
> > > > > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config set <var> <val>...
> > > > > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:admin_socket: invalid command
> > > > > > > 2025-02-27T16:07:55.092 DEBUG:teuthology.orchestra.run:got remote process result: 22
> > > > > > > ```
> > > > > > > 
> > > > > > > 
> > > > > > >     
> > > > > > >       
> > > > > > >     
> > > > > > > 
> > > > > > >       
> > > > > > >     
> > > > > > > 
> > > > > > >     
> > > > > > >   
> > > > > > > This^ should not happen. https://github.com/ceph/ceph/blob/main/qa/tasks/cephfs/test_mirroring.py#L898-L901 - This doesn't make the mirroring restarted and the command `fs mirror status` fails.
> > > > > > 
> > > > > > 
> > > > > > This means the mirroring commands are not registered. Is the mgr undergoing restarts?
> > > > > 
> > > > > 
> > > > > Yes, the mgr loads modules during that time. It doesn't seems responding to mirror commands during that time.
> > > > 
> > > > 
> > > > But `fs mirror ...` commands are run inside safe_while(), so it would retry till it times out. Did the test hit the max retry failure?
> > > 
> > > 
> > > Yes, it hit the max retry failure.
> > 
> > OK. In that case, if the timeout was due to ceph-mgr not responding to commands, then increasing the timeout could help in the interim. But, ceph-mgr not responding for such long is a cause of concern.
> 
> I can increase the timeout, but that's not the root cause. The root cause seems unrelated to mirroring.

Could you point to the log snippet that seems unrelated?

joscollin commented:
> > > > > > > > > @joscollin status of this? Seeing again in https://pulpito.ceph.com/vshankar-2025-02-27_13:15:04-fs-wip-vshankar-testing-20250227.075314-debug-testing-default-smithi/8158468/
> > > > > > > > 
> > > > > > > > 
> > > > > > > > @vshankar I don't see this failure coming in my latest tests, especially with the fix #61539.
> > > > > > > 
> > > > > > > 
> > > > > > > Could you explain how #61539 fixes this? I don't see an obvious relation.
> > > > > > 
> > > > > > 
> > > > > > I didn't say that #61539 fixes this issue. I meant, I don't see this failure coming in my latest tests.
> > > > > 
> > > > > 
> > > > > The "especially with fix ..." sounded like that PR fixes something under-the-hood.
> > > > > > > > Checking the logs, I see:
> > > > > > > > ```
> > > > > > > > 2025-02-27T16:07:49.928 WARNING:tasks.cephfs.test_mirroring:mirror daemon command with label "mirror status for fs: cephfs" failed: Command failed (mirror status for fs: cephfs) on smithi097 with status 22: 'ceph --admin-daemon /var/run/ceph/cephfs-mirror.asok fs mirror status cephfs@62'
> > > > > > > > 2025-02-27T16:07:54.929 INFO:teuthology.orchestra.run:Running command with timeout 30
> > > > > > > > 2025-02-27T16:07:54.930 DEBUG:teuthology.orchestra.run.smithi097:mirror status for fs: cephfs> ceph --admin-daemon /var/run/ceph/cephfs-mirror.asok fs mirror status cephfs@62
> > > > > > > > 2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:no valid command found; 10 closest matches:
> > > > > > > > 2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:0
> > > > > > > > 2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:1
> > > > > > > > 2025-02-27T16:07:55.090 INFO:teuthology.orchestra.run.smithi097.stderr:2
> > > > > > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:abort
> > > > > > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:assert
> > > > > > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config diff
> > > > > > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config diff get <var>
> > > > > > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config get <var>
> > > > > > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config help [<var>]
> > > > > > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:config set <var> <val>...
> > > > > > > > 2025-02-27T16:07:55.091 INFO:teuthology.orchestra.run.smithi097.stderr:admin_socket: invalid command
> > > > > > > > 2025-02-27T16:07:55.092 DEBUG:teuthology.orchestra.run:got remote process result: 22
> > > > > > > > ```
> > > > > > > > 
> > > > > > > > 
> > > > > > > >     
> > > > > > > >       
> > > > > > > >     
> > > > > > > > 
> > > > > > > >       
> > > > > > > >     
> > > > > > > > 
> > > > > > > >     
> > > > > > > >   
> > > > > > > > This^ should not happen. https://github.com/ceph/ceph/blob/main/qa/tasks/cephfs/test_mirroring.py#L898-L901 - This doesn't make the mirroring restarted and the command `fs mirror status` fails.
> > > > > > > 
> > > > > > > 
> > > > > > > This means the mirroring commands are not registered. Is the mgr undergoing restarts?
> > > > > > 
> > > > > > 
> > > > > > Yes, the mgr loads modules during that time. It doesn't seems responding to mirror commands during that time.
> > > > > 
> > > > > 
> > > > > But `fs mirror ...` commands are run inside safe_while(), so it would retry till it times out. Did the test hit the max retry failure?
> > > > 
> > > > 
> > > > Yes, it hit the max retry failure.
> > > 
> > > 
> > > OK. In that case, if the timeout was due to ceph-mgr not responding to commands, then increasing the timeout could help in the interim. But, ceph-mgr not responding for such long is a cause of concern.
> > 
> > 
> > I can increase the timeout, but that's not the root cause. The root cause seems unrelated to mirroring.
> 
> Could you point to the log snippet that seems unrelated?

I meant there's no reply for `fs mirror status` in the mgr logs. There's mgr respawn and loading modules.

vshankar commented:
> I meant there's no reply for `fs mirror status` in the mgr logs. There's mgr respawn and loading modules.

This is in the mirror daemon logs where the test (`test_mirroring_init_failure_with_recovery`) starts:

```
2025-02-27T16:06:58.858+0000 10ba7640 20 cephfs::mirror::Mirror handle_enable_mirroring: filesystem={fscid=60, fs_name=cephfs}, peers=, r=-2
2025-02-27T16:06:58.858+0000 10ba7640 -1 cephfs::mirror::Mirror handle_enable_mirroring: failed to initialize FSMirror for filesystem={fscid=60, fs_name=cephfs}: (2) No such file or directory
2025-02-27T16:06:58.858+0000 10ba7640 10 cephfs::mirror::ServiceDaemon: 0x12c194d0 add_or_update_fs_attribute: fscid=60
2025-02-27T16:06:58.858+0000 10ba7640 10 cephfs::mirror::ServiceDaemon: 0x12c194d0 schedule_update_status
2025-02-27T16:06:59.619+0000 113a8640 20 cephfs::mirror::Mirror update_fs_mirrors
2025-02-27T16:06:59.620+0000 113a8640  5 cephfs::mirror::Mirror update_fs_mirrors: filesystem={fscid=60, fs_name=cephfs} failed mirroring (failed: 1, blocklisted: 0)
2025-02-27T16:06:59.620+0000 113a8640 10 cephfs::mirror::Mirror disable_mirroring: init failed for filesystem={fscid=60, fs_name=cephfs}
2025-02-27T16:06:59.620+0000 113a8640 20 cephfs::mirror::Mirror schedule_mirror_update_task: scheduling fs mirror update (0x6ef87a0) after 2 seconds
2025-02-27T16:06:59.620+0000 10ba7640 20 cephfs::mirror::FSMirror ~FSMirror
2025-02-27T16:06:59.620+0000 10ba7640 10 cephfs::mirror::Mirror enable_mirroring: starting FSMirror: filesystem={fscid=60, fs_name=cephfs}
2025-02-27T16:06:59.620+0000 10ba7640 10 cephfs::mirror::ServiceDaemon: 0x12c194d0 add_or_update_fs_attribute: fscid=60
2025-02-27T16:06:59.620+0000 10ba7640 10 cephfs::mirror::ServiceDaemon: 0x12c194d0 schedule_update_status
2025-02-27T16:06:59.621+0000 10ba7640 20 cephfs::mirror::FSMirror init
2025-02-27T16:06:59.621+0000 10ba7640 20 cephfs::mirror::Utils connect: connecting to cluster=ceph, client=client.mirror, mon_host=
2025-02-27T16:07:00.087+0000 10ba7640 10 cephfs::mirror::Utils connect: using mon addr=172.21.15.97
2025-02-27T16:07:00.088+0000 157b0640 20 cephfs::mirror::ServiceDaemon: 0x12c194d0 update_status: 1 filesystem(s)
2025-02-27T16:07:00.090+0000 f3a4640 20 cephfs::mirror::ClusterWatcher handle_fsmap
2025-02-27T16:07:00.091+0000 f3a4640  5 cephfs::mirror::ClusterWatcher handle_fsmap: mirroring enabled=[], mirroring_disabled=[]
2025-02-27T16:07:00.091+0000 f3a4640  5 cephfs::mirror::ClusterWatcher handle_fsmap: peers added={}, peers removed={}
2025-02-27T16:07:00.406+0000 10ba7640 10 cephfs::mirror::Utils connect: connected to cluster=ceph using client=client.mirror
2025-02-27T16:07:00.431+0000 10ba7640 20 cephfs::mirror::Utils mount: filesystem={fscid=60, fs_name=cephfs}
2025-02-27T16:07:00.940+0000 f3a4640 20 cephfs::mirror::ClusterWatcher handle_fsmap
2025-02-27T16:07:00.942+0000 f3a4640  5 cephfs::mirror::ClusterWatcher handle_fsmap: mirroring enabled=[], mirroring_disabled=[{fscid=60, fs_name=cephfs}]
2025-02-27T16:07:00.943+0000 f3a4640 10 cephfs::mirror::ServiceDaemon: 0x12c194d0 remove_filesystem: fscid=60
2025-02-27T16:07:00.943+0000 f3a4640 10 cephfs::mirror::ServiceDaemon: 0x12c194d0 schedule_update_status
2025-02-27T16:07:00.943+0000 f3a4640 10 cephfs::mirror::Mirror mirroring_disabled: filesystem={fscid=60, fs_name=cephfs}
2025-02-27T16:07:01.621+0000 113a8640 20 cephfs::mirror::Mirror update_fs_mirrors
2025-02-27T16:07:01.944+0000 157b0640 20 cephfs::mirror::ServiceDaemon: 0x12c194d0 update_status: 0 filesystem(s)
2025-02-27T16:07:10.787+0000 10ba7640 -1 cephfs::mirror::Utils mount: mount error: (110) Connection timed out
2025-02-27T16:07:10.813+0000 10ba7640 20 cephfs::mirror::Mirror handle_enable_mirroring: filesystem={fscid=60, fs_name=cephfs}, peers=, r=-110
2025-02-27T16:07:10.813+0000 10ba7640 -1 cephfs::mirror::Mirror handle_enable_mirroring: failed to initialize FSMirror for filesystem={fscid=60, fs_name=cephfs}: (110) Connection timed out
2025-02-27T16:07:10.814+0000 10ba7640 10 cephfs::mirror::ServiceDaemon: 0x12c194d0 add_or_update_fs_attribute: fscid=60
2025-02-27T16:07:10.814+0000 f3a4640  5 cephfs::mirror::ClusterWatcher handle_fsmap: peers added={}, peers removed={}
2025-02-27T16:07:10.816+0000 f3a4640 20 cephfs::mirror::ClusterWatcher handle_fsmap
2025-02-27T16:07:10.817+0000 f3a4640  5 cephfs::mirror::ClusterWatcher handle_fsmap: mirroring enabled=[], mirroring_disabled=[]```

The mirror daemon hit a `-110` error (connection timeout) likely from the previous test and never recovered which caused the test `test_mirroring_init_failure_with_recovery` to fail. You should inspect the previous test to see what went wrong.

joscollin commented:
> > I meant there's no reply for `fs mirror status` in the mgr logs. There's mgr respawn and loading modules.
> 
> This is in the mirror daemon logs where the test (`test_mirroring_init_failure_with_recovery`) starts:
> 
> ```
> 2025-02-27T16:06:58.858+0000 10ba7640 20 cephfs::mirror::Mirror handle_enable_mirroring: filesystem={fscid=60, fs_name=cephfs}, peers=, r=-2
> 2025-02-27T16:06:58.858+0000 10ba7640 -1 cephfs::mirror::Mirror handle_enable_mirroring: failed to initialize FSMirror for filesystem={fscid=60, fs_name=cephfs}: (2) No such file or directory
> 2025-02-27T16:06:58.858+0000 10ba7640 10 cephfs::mirror::ServiceDaemon: 0x12c194d0 add_or_update_fs_attribute: fscid=60
> 2025-02-27T16:06:58.858+0000 10ba7640 10 cephfs::mirror::ServiceDaemon: 0x12c194d0 schedule_update_status
> 2025-02-27T16:06:59.619+0000 113a8640 20 cephfs::mirror::Mirror update_fs_mirrors
> 2025-02-27T16:06:59.620+0000 113a8640  5 cephfs::mirror::Mirror update_fs_mirrors: filesystem={fscid=60, fs_name=cephfs} failed mirroring (failed: 1, blocklisted: 0)
> 2025-02-27T16:06:59.620+0000 113a8640 10 cephfs::mirror::Mirror disable_mirroring: init failed for filesystem={fscid=60, fs_name=cephfs}
> 2025-02-27T16:06:59.620+0000 113a8640 20 cephfs::mirror::Mirror schedule_mirror_update_task: scheduling fs mirror update (0x6ef87a0) after 2 seconds
> 2025-02-27T16:06:59.620+0000 10ba7640 20 cephfs::mirror::FSMirror ~FSMirror
> 2025-02-27T16:06:59.620+0000 10ba7640 10 cephfs::mirror::Mirror enable_mirroring: starting FSMirror: filesystem={fscid=60, fs_name=cephfs}
> 2025-02-27T16:06:59.620+0000 10ba7640 10 cephfs::mirror::ServiceDaemon: 0x12c194d0 add_or_update_fs_attribute: fscid=60
> 2025-02-27T16:06:59.620+0000 10ba7640 10 cephfs::mirror::ServiceDaemon: 0x12c194d0 schedule_update_status
> 2025-02-27T16:06:59.621+0000 10ba7640 20 cephfs::mirror::FSMirror init
> 2025-02-27T16:06:59.621+0000 10ba7640 20 cephfs::mirror::Utils connect: connecting to cluster=ceph, client=client.mirror, mon_host=
> 2025-02-27T16:07:00.087+0000 10ba7640 10 cephfs::mirror::Utils connect: using mon addr=172.21.15.97
> 2025-02-27T16:07:00.088+0000 157b0640 20 cephfs::mirror::ServiceDaemon: 0x12c194d0 update_status: 1 filesystem(s)
> 2025-02-27T16:07:00.090+0000 f3a4640 20 cephfs::mirror::ClusterWatcher handle_fsmap
> 2025-02-27T16:07:00.091+0000 f3a4640  5 cephfs::mirror::ClusterWatcher handle_fsmap: mirroring enabled=[], mirroring_disabled=[]
> 2025-02-27T16:07:00.091+0000 f3a4640  5 cephfs::mirror::ClusterWatcher handle_fsmap: peers added={}, peers removed={}
> 2025-02-27T16:07:00.406+0000 10ba7640 10 cephfs::mirror::Utils connect: connected to cluster=ceph using client=client.mirror
> 2025-02-27T16:07:00.431+0000 10ba7640 20 cephfs::mirror::Utils mount: filesystem={fscid=60, fs_name=cephfs}
> 2025-02-27T16:07:00.940+0000 f3a4640 20 cephfs::mirror::ClusterWatcher handle_fsmap
> 2025-02-27T16:07:00.942+0000 f3a4640  5 cephfs::mirror::ClusterWatcher handle_fsmap: mirroring enabled=[], mirroring_disabled=[{fscid=60, fs_name=cephfs}]
> 2025-02-27T16:07:00.943+0000 f3a4640 10 cephfs::mirror::ServiceDaemon: 0x12c194d0 remove_filesystem: fscid=60
> 2025-02-27T16:07:00.943+0000 f3a4640 10 cephfs::mirror::ServiceDaemon: 0x12c194d0 schedule_update_status
> 2025-02-27T16:07:00.943+0000 f3a4640 10 cephfs::mirror::Mirror mirroring_disabled: filesystem={fscid=60, fs_name=cephfs}
> 2025-02-27T16:07:01.621+0000 113a8640 20 cephfs::mirror::Mirror update_fs_mirrors
> 2025-02-27T16:07:01.944+0000 157b0640 20 cephfs::mirror::ServiceDaemon: 0x12c194d0 update_status: 0 filesystem(s)
> 2025-02-27T16:07:10.787+0000 10ba7640 -1 cephfs::mirror::Utils mount: mount error: (110) Connection timed out
> 2025-02-27T16:07:10.813+0000 10ba7640 20 cephfs::mirror::Mirror handle_enable_mirroring: filesystem={fscid=60, fs_name=cephfs}, peers=, r=-110
> 2025-02-27T16:07:10.813+0000 10ba7640 -1 cephfs::mirror::Mirror handle_enable_mirroring: failed to initialize FSMirror for filesystem={fscid=60, fs_name=cephfs}: (110) Connection timed out
> 2025-02-27T16:07:10.814+0000 10ba7640 10 cephfs::mirror::ServiceDaemon: 0x12c194d0 add_or_update_fs_attribute: fscid=60
> 2025-02-27T16:07:10.814+0000 f3a4640  5 cephfs::mirror::ClusterWatcher handle_fsmap: peers added={}, peers removed={}
> 2025-02-27T16:07:10.816+0000 f3a4640 20 cephfs::mirror::ClusterWatcher handle_fsmap
> 2025-02-27T16:07:10.817+0000 f3a4640  5 cephfs::mirror::ClusterWatcher handle_fsmap: mirroring enabled=[], mirroring_disabled=[]```
> 
> The mirror daemon hit a `-110` error (connection timeout) likely from the previous test and never recovered which caused the test `test_mirroring_init_failure_with_recovery` to fail. You should inspect the previous test to see what went wrong.
> ```

I will check that. But it's not happening with every job failures in this ticket. For instance, I have (downloaded earlier) the entire logs of 
https://pulpito.ceph.com/vshankar-2024-09-18_11:34:04-fs-wip-vshankar-testing-20240917.043942-debug-testing-default-smithi/7911148, which doesn't have that error. But I'll check the jobs which has -110 error.

vshankar commented:
> I will check that. But it's not happening with every job failures in this ticket. For instance, I have (downloaded earlier) the entire logs of
> https://pulpito.ceph.com/vshankar-2024-09-18_11:34:04-fs-wip-vshankar-testing-20240917.043942-debug-testing-default-smithi/7911148, which doesn't have that error. But I'll check the jobs which has -110 error.

that's fine - at times such failures are seen only at times. We have the logs for this failure, so please RCA it.

joscollin commented:
> > I will check that. But it's not happening with every job failures in this ticket. For instance, I have (downloaded earlier) the entire logs of
> > https://pulpito.ceph.com/vshankar-2024-09-18_11:34:04-fs-wip-vshankar-testing-20240917.043942-debug-testing-default-smithi/7911148, which doesn't have that error. But I'll check the jobs which has -110 error.
> 
> that's fine - at times such failures are seen only at times. We have the logs for this failure, so please RCA it.

As I mentioned [here](https://github.com/ceph/ceph/pull/60052#issuecomment-2696754855), the [previous test](https://github.com/ceph/ceph/blob/main/qa/tasks/cephfs/test_mirroring.py#L865) also doesn't restart the mirroring and the `fs mirror status` command fails with `client.0 error registering admin socket command: (17) File exists`.

```
2025-01-09T13:44:49.366 DEBUG:teuthology.orchestra.run.smithi138:> sudo adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage timeout 120 ceph --cluster ceph fs snapshot mirror enable cephfs
2025-01-09T13:44:52.381 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.377+0000 7f1a4515d640 -1 client.0 error registering admin socket command: (17) File exists
2025-01-09T13:44:52.381 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.377+0000 7f1a4515d640 -1 client.0 error registering admin socket command: (17) File exists
2025-01-09T13:44:52.381 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.377+0000 7f1a4515d640 -1 client.0 error registering admin socket command: (17) File exists
2025-01-09T13:44:52.381 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.377+0000 7f1a4515d640 -1 client.0 error registering admin socket command: (17) File exists
2025-01-09T13:44:52.381 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.377+0000 7f1a4515d640 -1 client.0 error registering admin socket command: (17) File exists
2025-01-09T13:44:52.382 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.377+0000 7f1a42117640 -1 client.0 error registering admin socket command: (17) File exists
2025-01-09T13:44:52.382 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.377+0000 7f1a42117640 -1 client.0 error registering admin socket command: (17) File exists
2025-01-09T13:44:52.382 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.377+0000 7f1a42117640 -1 client.0 error registering admin socket command: (17) File exists
2025-01-09T13:44:52.382 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.377+0000 7f1a42117640 -1 client.0 error registering admin socket command: (17) File exists
2025-01-09T13:44:52.382 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.377+0000 7f1a42117640 -1 client.0 error registering admin socket command: (17) File exists
2025-01-09T13:44:52.396 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.392+0000 7f1a42117640 -1 client.0 error registering admin socket command: (17) File exists
2025-01-09T13:44:52.397 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.392+0000 7f1a42117640 -1 client.0 error registering admin socket command: (17) File exists
2025-01-09T13:44:52.397 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.392+0000 7f1a42117640 -1 client.0 error registering admin socket command: (17) File exists
2025-01-09T13:44:52.397 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.392+0000 7f1a42117640 -1 client.0 error registering admin socket command: (17) File exists
2025-01-09T13:44:52.397 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.392+0000 7f1a42117640 -1 client.0 error registering admin socket command: (17) File exists
2025-01-09T13:44:52.397 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.393+0000 7f1a4515d640 -1 client.0 error registering admin socket command: (17) File exists
2025-01-09T13:44:52.397 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.393+0000 7f1a4515d640 -1 client.0 error registering admin socket command: (17) File exists
2025-01-09T13:44:52.397 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.393+0000 7f1a4515d640 -1 client.0 error registering admin socket command: (17) File exists
2025-01-09T13:44:52.397 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.393+0000 7f1a4515d640 -1 client.0 error registering admin socket command: (17) File exists
2025-01-09T13:44:52.397 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.393+0000 7f1a4515d640 -1 client.0 error registering admin socket command: (17) File exists
2025-01-09T13:44:54.340 INFO:teuthology.orchestra.run.smithi138.stdout:{}
2025-01-09T13:45:04.357 INFO:teuthology.orchestra.run:Running command with timeout 30
2025-01-09T13:45:04.357 DEBUG:teuthology.orchestra.run.smithi138:mirror status for fs: cephfs> ceph --admin-daemon /var/run/ceph/cephfs-mirror.asok fs mirror status cephfs@2
2025-01-09T13:45:04.476 INFO:teuthology.orchestra.run.smithi138.stdout:{
2025-01-09T13:45:04.476 INFO:teuthology.orchestra.run.smithi138.stdout:    "rados_inst": "172.21.15.138:0/2820423427",
2025-01-09T13:45:04.476 INFO:teuthology.orchestra.run.smithi138.stdout:    "peers": {},
2025-01-09T13:45:04.476 INFO:teuthology.orchestra.run.smithi138.stdout:    "snap_dirs": {
2025-01-09T13:45:04.477 INFO:teuthology.orchestra.run.smithi138.stdout:        "dir_count": 0
2025-01-09T13:45:04.477 INFO:teuthology.orchestra.run.smithi138.stdout:    }
2025-01-09T13:45:04.477 INFO:teuthology.orchestra.run.smithi138.stdout:}
2025-01-09T13:45:04.484 DEBUG:tasks.cephfs.test_mirroring:command returned={
    "rados_inst": "172.21.15.138:0/2820423427",
    "peers": {},
    "snap_dirs": {
        "dir_count": 0
    }
}

```

May be, we need to increase the sleep, so that mirror daemon restarts in the mean time? Because it seems executing the last time^.

vshankar commented:
> > > I will check that. But it's not happening with every job failures in this ticket. For instance, I have (downloaded earlier) the entire logs of
> > > https://pulpito.ceph.com/vshankar-2024-09-18_11:34:04-fs-wip-vshankar-testing-20240917.043942-debug-testing-default-smithi/7911148, which doesn't have that error. But I'll check the jobs which has -110 error.\

No, that message is just a warning and not the underlying reason for the test case failure.

> > 
> > 
> > that's fine - at times such failures are seen only at times. We have the logs for this failure, so please RCA it.
> 
> As I mentioned [here](https://github.com/ceph/ceph/pull/60052#issuecomment-2696754855), the [previous test](https://github.com/ceph/ceph/blob/main/qa/tasks/cephfs/test_mirroring.py#L865) also doesn't restart the mirroring and the `fs mirror status` command fails with `client.0 error registering admin socket command: (17) File exists`.
> 
> ```
> 2025-01-09T13:44:49.366 DEBUG:teuthology.orchestra.run.smithi138:> sudo adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage timeout 120 ceph --cluster ceph fs snapshot mirror enable cephfs
> 2025-01-09T13:44:52.381 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.377+0000 7f1a4515d640 -1 client.0 error registering admin socket command: (17) File exists
> 2025-01-09T13:44:52.381 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.377+0000 7f1a4515d640 -1 client.0 error registering admin socket command: (17) File exists
> 2025-01-09T13:44:52.381 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.377+0000 7f1a4515d640 -1 client.0 error registering admin socket command: (17) File exists
> 2025-01-09T13:44:52.381 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.377+0000 7f1a4515d640 -1 client.0 error registering admin socket command: (17) File exists
> 2025-01-09T13:44:52.381 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.377+0000 7f1a4515d640 -1 client.0 error registering admin socket command: (17) File exists
> 2025-01-09T13:44:52.382 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.377+0000 7f1a42117640 -1 client.0 error registering admin socket command: (17) File exists
> 2025-01-09T13:44:52.382 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.377+0000 7f1a42117640 -1 client.0 error registering admin socket command: (17) File exists
> 2025-01-09T13:44:52.382 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.377+0000 7f1a42117640 -1 client.0 error registering admin socket command: (17) File exists
> 2025-01-09T13:44:52.382 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.377+0000 7f1a42117640 -1 client.0 error registering admin socket command: (17) File exists
> 2025-01-09T13:44:52.382 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.377+0000 7f1a42117640 -1 client.0 error registering admin socket command: (17) File exists
> 2025-01-09T13:44:52.396 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.392+0000 7f1a42117640 -1 client.0 error registering admin socket command: (17) File exists
> 2025-01-09T13:44:52.397 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.392+0000 7f1a42117640 -1 client.0 error registering admin socket command: (17) File exists
> 2025-01-09T13:44:52.397 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.392+0000 7f1a42117640 -1 client.0 error registering admin socket command: (17) File exists
> 2025-01-09T13:44:52.397 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.392+0000 7f1a42117640 -1 client.0 error registering admin socket command: (17) File exists
> 2025-01-09T13:44:52.397 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.392+0000 7f1a42117640 -1 client.0 error registering admin socket command: (17) File exists
> 2025-01-09T13:44:52.397 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.393+0000 7f1a4515d640 -1 client.0 error registering admin socket command: (17) File exists
> 2025-01-09T13:44:52.397 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.393+0000 7f1a4515d640 -1 client.0 error registering admin socket command: (17) File exists
> 2025-01-09T13:44:52.397 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.393+0000 7f1a4515d640 -1 client.0 error registering admin socket command: (17) File exists
> 2025-01-09T13:44:52.397 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.393+0000 7f1a4515d640 -1 client.0 error registering admin socket command: (17) File exists
> 2025-01-09T13:44:52.397 INFO:tasks.ceph.mgr.x.smithi138.stderr:2025-01-09T13:44:52.393+0000 7f1a4515d640 -1 client.0 error registering admin socket command: (17) File exists
> 2025-01-09T13:44:54.340 INFO:teuthology.orchestra.run.smithi138.stdout:{}
> 2025-01-09T13:45:04.357 INFO:teuthology.orchestra.run:Running command with timeout 30
> 2025-01-09T13:45:04.357 DEBUG:teuthology.orchestra.run.smithi138:mirror status for fs: cephfs> ceph --admin-daemon /var/run/ceph/cephfs-mirror.asok fs mirror status cephfs@2
> 2025-01-09T13:45:04.476 INFO:teuthology.orchestra.run.smithi138.stdout:{
> 2025-01-09T13:45:04.476 INFO:teuthology.orchestra.run.smithi138.stdout:    "rados_inst": "172.21.15.138:0/2820423427",
> 2025-01-09T13:45:04.476 INFO:teuthology.orchestra.run.smithi138.stdout:    "peers": {},
> 2025-01-09T13:45:04.476 INFO:teuthology.orchestra.run.smithi138.stdout:    "snap_dirs": {
> 2025-01-09T13:45:04.477 INFO:teuthology.orchestra.run.smithi138.stdout:        "dir_count": 0
> 2025-01-09T13:45:04.477 INFO:teuthology.orchestra.run.smithi138.stdout:    }
> 2025-01-09T13:45:04.477 INFO:teuthology.orchestra.run.smithi138.stdout:}
> 2025-01-09T13:45:04.484 DEBUG:tasks.cephfs.test_mirroring:command returned={
>     "rados_inst": "172.21.15.138:0/2820423427",
>     "peers": {},
>     "snap_dirs": {
>         "dir_count": 0
>     }
> }
> ```
> 
> May be, we need to increase the sleep, so that mirror daemon restarts in the mean time? Because it seems executing the last time^.

The mirror daemon should hit failed state, but `fs mirror status` returned a valid json (without state as failed). I think the disable mirroring at the start of the the test `test_mirroring_init_failure` has still fully processed by the mirror daemon and  this LOC in the test https://github.com/ceph/ceph/blob/main/qa/tasks/cephfs/test_mirroring.py#L873 returns successfully, rather than actually checking the failed state. And after this, the test ends without running the rest of the test case!

That's weird. Do you too see this? @joscollin

joscollin commented:
> see

Yes, I've pasted that log in my previous comment. It returns because there's no `state` in the json. I think the command fails for 20 times, as the mirroring is not yet restarted or the restart is in progress. @vshankar

joscollin commented:
@vshankar 
Updated the PR. Hope that would prevent this issue.

vshankar commented on qa/tasks/cephfs/test_mirroring.py:
```diff
@@ -864,21 +864,17 @@ def test_mirroring_init_failure(self):
+                res = self.mirror_daemon_command(f'mirror status for fs: {self.primary_fs_name}',
+                                                 'fs', 'mirror', 'status', f'{self.primary_fs_name}@{self.primary_fs_id}')
+                if 'state' in res and res['state'] == "failed":
+                    break
```
Why `break` and not `return`?

joscollin replied:
dropped it.

vshankar commented:
> > see
> 
> Yes, I've pasted that log in my previous comment. It returns because there's no `state` in the json. I think the command fails for 20 times, as the mirroring is not yet restarted or the restart is in progress. @vshankar

While the check you introduced in the latest revision now checks for `state` to be failed, I don't understand how that fixes the issue. Without this current change, the test code would consider a non-failed state as success under safe_while(), but that does not explain the reason why the test failed.

joscollin commented:
> > > see
> > 
> > 
> > Yes, I've pasted that log in my previous comment. It returns because there's no `state` in the json. I think the command fails for 20 times, as the mirroring is not yet restarted or the restart is in progress. @vshankar
> 
> While the check you introduced in the latest revision now checks for `state` to be failed, I don't understand how that fixes the issue. Without this current change, the test code would consider a non-failed state as success under safe_while(), but that does not explain the reason why the test failed.

Dropped those changes and updated the PR.

@vshankar None of these fixes I've provided explains/fixes the root cause of this issue (I'm aware of that). This PR was marked as a draft. I don't know why this PR started getting comments before I investigate, provided the right fix and mark it ready for review. So I was just answering the questions from https://github.com/ceph/ceph/pull/60052#issuecomment-2696754855 onwards. This shouldn't be a high priority, as this failure is not seen in the latest main branch atm.

Could you please give me some time to look deeper into this issue (why mgr is not responding to the commands and in the latest main it is responding) and then we will discuss after providing a fix?

vshankar commented:
Sure, I was interested in seeing the proposed fix even if it was a draft, just to provide initial comments. Tag me when it's ready for review.
